### DISCUSSION TITLE
Shadcn's new Sidebar covering (overlapping) any header or navbar

### DISCUSSION THREAD

**Discussion by surafel-99** [shadcn-ui/ui#5636] [General]
## Sidebar overlapping with header

Shadcn's new Sidebar component seems to cover any header or navbar. I was playing with the z index but can't make it work no matter how I try.

my current layout structure looks like this:
```
                  <html lang="en">
                    <body className={inter.className}>
        
                            <ThemeProvider
                            attribute="class"
                            defaultTheme="system"
                            enableSystem
                            disableTransitionOnChange 
                        >
                              
                              <Navbar />
                            
                              <FavoriteModal />
                              <ToastProvider />
                            
                              <SidebarProvider>
                                  <AppSidebar />
                                  <main className="w-full">
                                    <SidebarTrigger />
                                    {children}
                                  </main>
                              </SidebarProvider>
            
                            </ThemeProvider>
                  </body>
            </html>
```
                
                
 I tried to add z index in the  Navbar component and Sidebar Component which is found in the app-sidebar.tsx file. But it didn't work.

```
// navbar in the layout.tsx
 <Navbar className="z-10" />


// app-sidebar.tsx file
 <Sidebar className="z-0">
   //sidebar content
</Sidebar>
```

**Comment by hydrokat** [comment]
Have you tried putting the overlapped elements inside the sidebar provider? I am currently experimenting with it as well and it seems that anything outside the sidebarprovider gets overlapped.

I have done it this way

```
//root layout
<html lang="en">
      <body className={inter.className}>
        <SidebarProvider>
          <div className="bg-background">{children}</div>
        </SidebarProvider>
      </body>
    </html>
```

```
// page layout
<div className="flex flex-row">
      <CustomSidebar menuItems={sidebarMenuItems} />
      <div className="p-4 flex-auto">
        <TopMenu topMenuItems={topMenuItems} /> <!~- the sidebar trigger is inside this -->
        {children}
      </div>
    </div>
```

**Reply by ViniciusPilati** [reply]
I made this attempt, I added the provider to the root of my application, it worked the same in my opinion

**Comment by GabrielEValenzuela** [comment]
Hello! How are you? 

Sorry for this probably stupid question, but I don't fully know about front topis 😅 

I have a similar issue with Sidebar component integrating ReacFlow, I get this 

![image](https://github.com/user-attachments/assets/3078efd6-29ca-40d4-ab38-b250e254bf15)


I tried to modify Tailwind, CSS, and make a div, but nothing happened. If I remove the 

```css
group/sidebar-wrapper flex min-h-svh w-full has-[[data-variant=inset]]:bg-sidebar
```

All works fine.

I change the order of in/out but, it doesn't work either. Could you solved?


Thanks in advance!

**Comment by ViniciusPilati** [comment]
I still have the same problem, the sidebar, when opened, covers the header of my application, even though it is a parent element of the sidebar, has anyone found a solution for this?

**Reply by kdv1995** [reply]
Hi, it seems you need this approach to use: https://github.com/shadcn-ui/ui/issues/5629

But please, remember about one stuffs in the guy's code; Don't forget to add "overflow-hidden" to the body. Otherwise, would waste your time to understand why it creates scroll on whole body; cheers

**Comment by GabrielEValenzuela** [comment]
Hi @ViniciusPilati2005 !

Maybe this helps you

##### page.tsx
```ts
'use client'

import React from 'react'
import { Provider } from 'react-redux'
import { store } from '@/lib/store'
import { SidebarProvider } from '@/components/ui/sidebar'
import AppContent from '@/components/AppContent'

const App: React.FC = () => {
  return (
    <Provider store={store}>
      <SidebarProvider>
        <AppContent />
      </SidebarProvider>
    </Provider>
  )
}

export default App
```


-----------------


##### AppContent.tsx
```ts
return (
    <>
      <SidebarInset className="flex-grow relative">
        <Toolbar />
        <Canvas />
      </SidebarInset>
      {isSidebarVisible && <ElementSidebar element={element} />}
      <Toast />
    </>
  )
```

**Reply by ViniciusPilati** [reply]
Não testei isso, mas como minha barra estava sobrepondo o cabeçalho, funcionou, para mim, definir um tamanho fixo no header e aplicar margem na sidebar: 

` <Sidebar className="overflow-hidden mt-[85px] h-auto">`

**Comment by minhvu7896** [comment]
Same issues, It covering `<Header />` and `<Footer />` components, anyone fixed it?

**Comment by lassounski** [comment]
I was having the same issue over here.

The behavior is a feature since the side bar is fixed while you scroll.

If you're are okay with the sidebar rolling with the rest of the page you can try this:

Find the keyword **fixed** in src/components/ui/sidebar.tsx and replace it for **absolute**.

**Reply by kdv1995** [reply]
https://github.com/shadcn-ui/ui/issues/5629

**Comment by Joshua10052000** [comment]
I also encountered the same problem, I just styled the `<SidebarHeader />` how I will style the header and the outcome is actually great.
![Untitled](https://github.com/user-attachments/assets/97ddd38d-306c-40a3-b647-c101765f89d7)


Here's how I did it. 

```
import {
  Sidebar,
  SidebarContent,
  SidebarHeader,
  SidebarMenu,
  SidebarMenuButton,
  SidebarMenuItem,
  SidebarProvider,
  SidebarRail,
} from "@/components/ui/sidebar";
import { Input } from "@/components/ui/input";
import { Button } from "@/components/ui/button";
import { HomeIcon, MountainIcon, SearchIcon } from "lucide-react";
import { AppQueryProvider } from "@/components/globals/app-query-provider";
import { AppThemeProvider } from "@/components/globals/app-theme-provider";
import * as React from "react";
import "@/styles/globals.css";

interface RootLayoutProps {
  children: React.ReactNode;
}

const RootLayout: React.FC<RootLayoutProps> = ({ children }) => {
  return (
    <html lang="en" suppressHydrationWarning>
      <head />
      <body className="bg-background font-sans antialiased">
        <AppThemeProvider>
          <AppQueryProvider>
            <SidebarProvider>
              <Sidebar collapsible="icon">
                <SidebarHeader className="fixed inset-x-0 top-0 flex h-20 flex-row items-center justify-between border-b bg-background px-4 md:px-8">
                  <section>
                    <MountainIcon />
                  </section>
                  <section className="relative max-w-md flex-1">
                    <Input className="pr-9" placeholder="Search..." />
                    <Button
                      className="absolute right-0 top-0"
                      size="icon"
                      variant="ghost"
                    >
                      <SearchIcon />
                    </Button>
                  </section>
                  <section>
                    <Button>Sign In</Button>
                  </section>
                </SidebarHeader>
                <SidebarContent className="pt-20">
                  <SidebarMenu>
                    {[{ href: "/", label: "Home" }].map((link) => {
                      return (
                        <SidebarMenuItem key={link.href}>
                          <SidebarMenuButton>
                            <HomeIcon />
                            <span>{link.label}</span>
                          </SidebarMenuButton>
                        </SidebarMenuItem>
                      );
                    })}
                  </SidebarMenu>
                </SidebarContent>
                <SidebarRail />
              </Sidebar>
              <main className="min-h-screen pt-20">{children}</main>
            </SidebarProvider>
          </AppQueryProvider>
        </AppThemeProvider>
      </body>
    </html>
  );
};

export default RootLayout;
```

**Comment by kdv1995** [comment]
If needed help, i could help u

**Comment by piyush97** [comment]
Any update/fix for this?

**Comment by ezzabuzaid** [comment]
This fixed it for me.

replace all instances of `w-[--sidebar-width]` with `w-[var(--sidebar-width)]`

**Reply by dylanjcastillo** [reply]
You're the 🐐. Thank you!

**Reply by nolios** [reply]
The only drawback with this approach is that the sidebar toggle doesn’t close the sidebar on larger displays.

To work around that, I hide the toggle with `md:hidden`. On desktop, users generally don’t need to collapse the sidebar, so this feels like an acceptable trade-off.

Here’s the change I made in `sidebar.tsx`:
<img width="827" height="229" alt="image" src="https://github.com/user-attachments/assets/9a931aa9-88a4-4823-952b-f975d2fcf0de" />

This is essentially the reverse of Shadcn’s `hidden md:block` in `sidebar.tsx`:
<img width="1198" height="420" alt="image" src="https://github.com/user-attachments/assets/51b55304-627c-4235-9baf-d05597ce08ba" />

**Reply by Tantumonium** [reply]
Ugh, the HOURS I've spent trying to resolve this... why couldn't I have found this solution sooner?!

**Reply by shubhamtak007** [reply]
I'm not sure if the same issue will come up in Next.js, but I faced it in React and Vite, and replacing the className fixed it.

**Reply by nulldoubt** [reply]
You're the GOAT :fire:

**Reply by yonaries** [reply]
took me fucking hours to find the solution. thank you🔥

**Comment by tiagosilvam** [comment]
a solution here:

`MySidebar.tsx`
```tsx
export const MySidebar = ({ ...props }: ComponentProps<typeof Sidebar>) => {
  return (
    <Sidebar {...props}>
        {/* Your sidebar components here */}
    </Sidebar>
  );
};
```

`layout.tsx`
```tsx
...
<SidebarProvider>
  <div className="flex flex-col w-full">
    <MyHeader className="min-h-16" />
    <div className="flex">
      <MySidebar className="relative max-h-[calc(100dvh-64px)]" />
      <SidebarInset>
        {children}
      </SidebarInset>
    </div>
  </div>
<SidebarProvider>
...
```

**Comment by DenysVuika** [comment]
end of 2025, still broken

**Comment by octavaliente** [comment]
Today I just ran a similar issue, the Sidebar was overlaping the main content.

I managed to fix it by modifying `components/ui/sidebar.tsx`. In my case, the CSS variable wasn't being interpreted correctly without `var()`.

To solve the fix, I replaced: 
```diff
- width: "--sidebar-width",
+ width: "var(--sidebar-width)"
```
and
```diff
- width: (--sidebar-width-icon)
+ width: var(--sidebar-width-icon)
```
Lines: 205, 235, 240, 245 and 252.

This solved the overlapping immediately. Hope this helps!

**Comment by Toshimichi0915** [comment]
For me, adding `min-w-0` to the main content fixed the problem.

Before
```ts
<SidebarProvider defaultOpen={user != null}>
  <SidebarInset>
    <AppHeader />
    {children}
    <AppFooter />
  </SidebarInset>
  <AppSidebar user={user} />
  <Toaster />
</SidebarProvider>
```

<img width="324" height="69" alt="image" src="https://github.com/user-attachments/assets/53629747-cbc9-4148-a44e-33d66a4ba244" />

After
```ts
<SidebarProvider defaultOpen={user != null}>
  <SidebarInset className="min-w-0">
    <AppHeader />
    {children}
    <AppFooter />
  </SidebarInset>
  <AppSidebar user={user} />
  <Toaster />
</SidebarProvider>
```

<img width="318" height="80" alt="image" src="https://github.com/user-attachments/assets/21b1f906-6adf-49a5-a12b-f33403b9a26e" />

Maybe, this solution won't work for some people. But it's worth a try if you have horizontally long content.
